### PR TITLE
Provide token information with XML parsed data

### LINKED ISSUE #107: Provide hover to parsed tokens
A use case where it might be useful to provide hover to parsed tokens is that user can check if a token exists in a context (rather than misspelled), and quickly show its definition (from assignment or function parameter).

### LINKED ISSUE #120: Provide definition to parsed tokens
Similar to #107, we can provide definition to parsed tokens using the same method.

### LINKED ISSUE #122: Provide documentHighlight to parsed tokens
Similar with #107 and #120, `textDocument/documentHighlight` can be provided using XML parse data.

### PR DESCRIPTION
**Pull request by renkun-ken**
Closes #107
Closes #120 
Closes #122

* Symbol last definition:

image

image

image

image

* Function argument documentation

image

* Document highlight: show all token occurrences (rather than text occurrences which may match partially)

image

### CODE REVIEW

**Comment by renkun-ken** [comment]
@randy3k, this PR works well for me now. Please take a look.

**Comment by randy3k** [comment]
Wow, this is awesome. Good job. I will take a look probably tomorrow.

**Comment by renkun-ken** [comment]
758208f implements token definition using the same method suggested at #120.

**Comment by randy3k** [comment]
Thank you for working on it. I guess we should publish another CRAN release soon. @renkun-ken do you have anything else in your mind that you would like to include in the next release? Otherwise, I suggest freezing the master branch after merging this PR and we could evaluate if there are any bugs before we submit the package to CRAN (other development could still continue, but they won't be merged soon).

In the long term, we would need to separate the `develop` branch and the `master` branch. CRAN maintainers do not like frequent submission so we better make sure our releases are free of any obvious bugs.

**Comment by renkun-ken** [comment]
@randy3k, I agree. There has been a lot of changes and new features to include in a release. I'll do more testing with this PR and all new features in general and make sure the release to be submitted to CRAN is good enough for production use.

**Comment by renkun-ken** [comment]
@randy3k This PR works well for me and is ready for merge.

**Comment by renkun-ken** [comment]
@randy3k This PR works well and is ready to merge now. Please feel free to let me know if there's anything I can improve in this PR.

**Comment by renkun-ken** [comment]
As I commented at https://github.com/REditorSupport/languageserver/pull/82#issuecomment-541017445, I'd like to see how `parse_expr` could be moved toward XML approach. Hopefully, it will allow local function to have signature and scope completion items to have correct kind (either function or variable).

File: R/utils.R [hunk 1]
```diff
@@ -285,3 +287,30 @@ convert_doc_to_markdown <- function(doc) {
 convert_doc_string <- function(doc) {
     paste0(convert_doc_to_markdown(doc), collapse = " ")
 }
+
+#' @importFrom glue glue
+#' @import xml2
+NULL
+
```
Comment (randy3k) [inline R/utils.R]: I would merge this as is now. Would you move the import statements to https://github.com/REditorSupport/languageserver/blob/99ee77105407fa9ac7248626552d4b9f75e070c1/R/languageserver.R#L2

**Comment by randy3k** [comment]
Oops, there is a gitch of github which posted the approval twice.

**Comment by renkun-ken** [comment]
Thanks!

**Pull request merged**